addons/web/static/tests/views/form_tests.js

```diff
@@ -5815,6 +5815,53 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('form view is not broken if save failed in readonly mode on field changed', function(assert) {
+        assert.expect(5);
+
+        var failFlag = false;
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<header><field name="trululu" widget="statusbar" clickable="True"/></header>' +
+                '</form>',
+            res_id: 1,
+            mockRPC: function (route, args) {
+                if (args.method === 'write' && failFlag) {
+                    assert.step('fail_write');
+                    return $.Deferred().reject();
+                } else if (args.method === 'write') {
+                    assert.step(args.method);
+                }
+                return this._super.apply(this, arguments);
```

Comment:
i would assert all RPCs, so that you can check that it reloads correctly

Reply:
I did it previously but I need to add many calls in verifySteps so later I checking for only those url which is our concern, let me add it again